plugins/modules/docker_swarm_service.py

```diff
@@ -1773,6 +1797,10 @@ def compare(self, os):
             force_update = True
         if self.init is not None and self.init != os.init:
             differences.add('init', parameter=self.init, active=os.init)
+        if self.cap_add is not None and self.cap_add != os.cap_add:
```

Comment:
Lists should be compared with `has_list_changed`. It also takes care of sorting (I assume the order of the capabilities does not matter; if it does, add `sort_lists=False`):
```suggestion
        if has_list_changed(self.cap_add, os.cap_add):
```